go/result_set.go

```diff
@@ -0,0 +1,594 @@
+		if err != nil {
+			return nil, err
+		}
+		nodeList = append(nodeList, *dst)
+		// determine direction
+		stepType := step.Type
+		if stepType > 0 {
+			segStartNode = *src
+			segEndNode = *dst
+			segType = stepType
+		} else {
+			segStartNode = *dst // switch with src
+			segEndNode = *src
+			segType = -stepType
+		}
+		edge = &nebula.Edge{
+			Src:     []byte(segStartNode.GetID()),
+			Dst:     []byte(segEndNode.GetID()),
+			Type:    segType,
+			Name:    step.Name,
+			Ranking: step.Ranking,
+			Props:   step.Props,
+		}
+		relationship, err := genRelationship(edge)
+		if err != nil {
+			return nil, err
+		}
+		relationshipList = append(relationshipList, *relationship)
+
+		segList = append(segList, segment{
+			startNode:    segStartNode,
+			relationship: *relationship,
+			endNode:      segEndNode,
+		})
+		src = dst
+	}
+	return &PathWrapper{
+		nodeList:         nodeList,
+		relationshipList: relationshipList,
+		segments:         segList,
+	}, nil
+}
+
+// Return all values in the given column
+func (res ResultSet) GetColValues(colName string) ([]*nebula.Value, error) {
+	var valList []*nebula.Value
+	for _, record := range res.records {
+		val, err := record.GetColValue(colName)
+		if err != nil {
+			return nil, err
+		}
+		valList = append(valList, val)
+	}
+	return valList, nil
+}
+
+// Return all values in the given row
+func (res ResultSet) GetRowValues(index int) ([]*nebula.Value, error) {
+	if err := checkIndex(index, res.records); err != nil {
+		return nil, err
+	}
+	return res.records[index].row.Values, nil
+}
+
+func (res ResultSet) GetColNames() []string {
+	return res.columnNames
+}
+
+func (res ResultSet) GetErrorCode() graph.ErrorCode {
+	return res.errorCode
+}
+
+func (res ResultSet) GetErrorMsg() []byte {
+	return res.errorMsg
+}
+
+func (res ResultSet) GetRecords() []Record {
+	return res.records
+}
+
+func (res ResultSet) IsSucceed() bool {
+	return res.GetErrorCode() == graph.ErrorCode_SUCCEEDED
+}
+
+// Return the value in the row using row index or colname
+func (record Record) GetColValue(key interface{}) (*nebula.Value, error) {
+	// Get value by column name
+	if reflect.TypeOf(key).String() == "string" {
+		// Get index
+		index, err := record.getIndexbyColName(key.(string))
+		if err != nil {
+			return nil, err
+		}
+		return record.row.Values[index], nil
+	}
+	// Get value by row index
+	if reflect.TypeOf(key).String() == "int" {
+		index := key.(int)
+		// TODO: make this check a function
+		if err := checkIndex(index, record.row.Values); err != nil {
+			return nil, err
+		}
+		return record.row.Values[index], nil
+	}
+	return nil, fmt.Errorf("Failed to get Value: requested coloumn name or index is invalid, the parameter shuold be int or string")
+}
+
+/*
+	Return a Node if the value at key is a vertex.
+	Key is column name being look up if key is a String.
+	If key is an int, key is the index of row.
+*/
+func (record Record) AsNode(key interface{}) (*Node, error) {
```

Comment:
directly to define two functions.
and `the string param` function to call `int param function`